source/Parser.php

```diff
@@ -183,29 +162,34 @@ private function getCodeWithMacros($code)
 
     public function format($code)
     {
-        $file = tempnam(sys_get_temp_dir(), "pre");
-        file_put_contents($file, $code);
+        $path = __DIR__;
+        $code = $this->addOpeningTag(trim($code));
+        $encoded = base64_encode($code);
 
-        $path = realpath(__DIR__ . "/../node_modules/prettier/bin-prettier.js");
-        exec("{$path} --write --parser=php {$file} 1> /dev/null 2> /dev/null", $output);
+        $command = "node -e '
+            const atob = require(\"atob\")
+            const prettier = require(\"prettier\")
 
-        $code = file_get_contents($file);
-        unlink($file);
+            prettier.resolveConfig(\"{$path}\").then(options => {
+                try {
+                    const formatted = prettier.format(atob(\"{$encoded}\").trim(), options)
+                    console.log(formatted)
+                } catch (e) {}
+            })
+        '";
 
-        return $code;
-    }
+        exec($command, $output);
```

Comment:
This makes the assumption that the current working directory is in the pre folder. Since that's not the case when this is a dependency, you'd need to store the current working directory. 

I basically changed `exec` to:

```php
        $cwd = getcwd();
        chdir(__DIR__);
        exec($command, $output);
        chdir($cwd);
```

Otherwise node will node be able to find `node_modules` and will error out...